app/ide-desktop/client/tests/localWorkflow.spec.ts

```diff
@@ -60,22 +80,15 @@ electronTest('Local Workflow', async ({ page, app, projectsDir }) => {
   await expect(writeNode.locator('.TableVisualization')).toContainText('output_ensodryrun')
 
   expect(await fs.readdir(PROJECT_PATH)).not.toContain(OUTPUT_FILE)
-
+  await expect(page.locator('.GraphEditor .GraphNode.pending')).toHaveCount(0)
   // Press `Write once` button.
   await writeNode.locator('.More').click()
   await writeNode.getByTestId('recompute').click()
+  await page.mouse.move(0, 0) // Avoid showing a tooltip
+  await expect(page.locator('.GraphEditor .GraphNode.pending')).toHaveCount(0)
 
   // Check that the output file is created and contains expected text.
-  try {
-    await expect(writeNode.locator('.TableVisualization')).toContainText(OUTPUT_FILE)
-  } catch {
-    // TODO[ao]
-    // The above check is flaky, because sometimes the additional engine run overrides node output back to "dry run".
-    // To confirm if this should be expected.
-    console.error(
-      'Didn\'t see the visualization update after "Write once" action; assuming it\'s already done',
-    )
-  }
+  await expect(writeNode.locator('.TableVisualization')).toContainText(OUTPUT_FILE)
```

Comment:
Given the above, I still think this check is flaky and keep the try/catch.

Reply:
I don't think this is flaky anymore, as I didn't observe that happening. If there is a problem, we have to fix it instead of masking it out through this catch, because it influences the snapshot anyway and would cause instability there.